grid: migrate upstream changes for compatibility with 417 clay + kiln

tomholford opened the pull request:
# Context

This resolves https://github.com/tloncorp/landscape-apps/issues/1271 by migrating the frontend [changes](https://github.com/urbit/urbit/compare/master...next/417) made in the urbit/urbit repo to support the [kelvin 417 clay + kiln changes](https://roadmap.urbit.org/project/agents-in-clay).

# Changes

- import commits from urbit/urbit to integrate this repo with the clay + kiln changes for arvo 417
- replace usage of Vats with Pikes
- breaking: remove rendered `aeon` from system update view (the value was previously exposed via Vat, no longer exposed via Pike)

# Preview

## Before

![image.png](https://images.zenhubusercontent.com/6245cc0024d90711bcf4ea06/f9dbbd28-280e-4b68-8b84-1c0407ae11f7)

## After

![image](https://user-images.githubusercontent.com/16504501/205193591-a924a3bb-fa4e-4327-9c52-afb64f982ccb.png)

![image](https://user-images.githubusercontent.com/16504501/205193669-2f8af97a-eec2-4010-8048-5dd1df15bc9d.png)


# TODO

- [x] resolve remaining merge conflicts
- [x] publish `@urbit/api` package and point the package.json to updated version

jamesacklin commented:
Looks good. I can't tell from the diff, but have we decremented the kelvin version in the docket as well?

patosullivan reviewed (approved):
Nice, code looks good and runs well locally.

arthyn commented on ui/src/preferences/about-system/AboutSystem.tsx:
```diff
@@ -45,15 +43,6 @@ export const AboutSystem = () => {
             <p>
               Version {gardenCharge?.version} ({hash})
             </p>
-            {systemBlocked && (
-              <p>
-                Aeon {aeon}{' '}
-                <span className="text-orange-500 mx-4 space-x-2">
-                  <span>&mdash;&gt;</span> <span>/</span> <span>&mdash;&gt;</span>
-                </span>{' '}
-                Aeon {nextAeon}
-              </p>
-            )}
```
is there no replacement for this in the pike system?

tomholford replied:
Here is the schema for Pike:
```ts
export interface Pike {
  /**
   * Hash of the desk, rendered as `@uv`
   *
   * @remarks
   * Equivalent to
   * ```hoon
   * .^(@uv %cz /=desk=)
   * ```
   */
  hash: string;
  sync: {
    /**
     * Source desk for this Pike
     */
    desk: string;
    /**
     * Source ship for this Pike
     */
    ship: string;
  } | null;
  /**
   *  {@link Weft}s associated with this Pike
   */
  wefts: Weft[];
  /**
   * how live is this pike?
   * 
   * live - app is running
   * held - app is not running, but is trying to run. this state can be entered
   * in two main ways: 
   *   - when installing an app but it hasn't finished downloading (or it did 
   *     but failed to install for some reason)
   *   - when user forced a kelvin upgrade by suspending desks.
   * dead - app is not running
   */
  zest: "live" | "dead" | "held";
}
```

And here's a Weft:
```ts
/**
 * A component's kelvin version
 */
export interface Weft {
  /**
   *  Name of the component
   *
   *  @remarks
   *  Usually %zuse, %hoon, or %lull
   */
  name: string;
  /**
   * Kelvin version
   *
   */
  kelvin: number;
}
```

tomholford replied:
So looking at the new schema posted in previous comment, not seeing any `aeon` fields.

arthyn commented on ui/src/preferences/about-system/UpdatePreferences.tsx:
```diff
@@ -1,81 +1,23 @@
+  const { toggleInstall } = useKilnState();
+  
   return (
-    <div className="inner-section space-y-8 relative">
```
This should wrap around the system updates source setter. Also not sure about the title changing here @urcades

tomholford replied:
Good call, will make the change

tomholford replied:
Added to both instances of `SourceSetter`, looks much better:

![image](https://user-images.githubusercontent.com/16504501/205400919-03a5ce38-5cde-4e8d-860c-f27e000831e1.png)

![image](https://user-images.githubusercontent.com/16504501/205400938-4c0eb6d3-5fe2-4800-b096-41293a6df6fe.png)

arthyn reviewed (changes requested):
One small change and one question

tomholford commented:
@arthyn Addressed feedback on the `SourceSetter` container. As for `aeon`, posted the type corresponding to what is returned by the backend to confirm that the property is not being returned.

arthyn reviewed (approved):
lgtm!
